languagetool-language-modules/ga/src/main/java/org/languagetool/rules/ga/SpecificCaseRule.java

```diff
@@ -0,0 +1,118 @@
+/* LanguageTool, a natural language style checker 
+ * Copyright (C) 2019 Daniel Naber (http://www.danielnaber.de)
+ * 
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301
+ * USA
+ */
+package org.languagetool.rules.ga;
+
+import org.languagetool.AnalyzedSentence;
+import org.languagetool.AnalyzedTokenReadings;
+import org.languagetool.JLanguageTool;
+import org.languagetool.rules.*;
+import org.languagetool.tagging.ga.Utils;
+
+import java.io.InputStream;
+import java.util.*;
+
+/**
+ * A rule that matches words which are complex and suggests easier to understand alternatives. 
+ * @since 4.8
+ */
+public class SpecificCaseRule extends Rule {
+  
+  private static final Set<String> phrases = new HashSet<>(loadPhrases("/ga/specific_case.txt"));
+  private static int maxLen;
+
+  private static List<String> loadPhrases(String path) {
+    List<String> l = new ArrayList<>();
+    InputStream file = JLanguageTool.getDataBroker().getFromResourceDirAsStream(path);
+    try (Scanner scanner = new Scanner(file, "UTF-8")) {
+      while (scanner.hasNextLine()) {
+        String line = scanner.nextLine().trim();
+        if (line.isEmpty() || line.startsWith("#")) {
+          continue;
+        }
+        int parts = line.split(" ").length;
+        maxLen = Math.max(parts, maxLen);
+        l.add(line.trim());
+      }
+    }
+    return l;
+  }
+
+  private static final Map<String,String> lcToProperSpelling = new HashMap<>();
+  static {
+    for (String phrase : phrases) {
+      lcToProperSpelling.put(phrase.toLowerCase(), phrase);
+    }
+  }
+
+  public SpecificCaseRule(ResourceBundle messages) {
+    super(messages);
+    super.setCategory(Categories.CASING.getCategory(messages));
+    setLocQualityIssueType(ITSIssueType.Misspelling);
+    addExamplePair(Example.wrong("Rugadh agus tógadh i <marker>mbéal Feirste</marker> é."),
+                   Example.fixed("Rugadh agus tógadh i <marker>mBéal Feirste</marker> é."));
+  }
+
+  @Override
+  public final String getId() {
+    return "GA_SPECIFIC_CASE";
+  }
+
+  @Override
+  public String getDescription() {
+    return "Checks upper/lower case spelling of some proper nouns";
+  }
+
+  @Override
+  public RuleMatch[] match(AnalyzedSentence sentence) {
```

Comment:
Is this a 1:1 copy? In that case, we should introduce a class in core that all languages can derive from, to avoid duplication.

Reply:
Not exactly, because the case checking for Irish is weird. For languages with normal casing, I think you could go with a class in core.